src/game/piece_move.rs

```diff
@@ -0,0 +1,138 @@
+use super::piece_type::*;
+
+fn validate_moves(moves: Vec<(u8, u8)>) -> Vec<(u8, u8)> {
+    let mut valid_moves: Vec<(u8, u8)> = [].to_vec();
+    for m in moves.iter() {
+        let (m1, m2) = *m;
+        if (m1 > 1 && m1 < 8) && (m2 < 8) {
```

Comment:
Please review this logic to ensure moves for all piece types are validated correctly.

Reply:
It's just the pawn so far but now that I'm looking at it I'm wondering about `m1>1`. Will ponder this algorithm a bit more

Reply:
yeah i found a bug in this implementation, so I've added some more tests (one of which now breaks). I'll send this back to @aoshi321 . Thanks for catching!

Reply:
this is now resolved so I'm merging